staging/src/k8s.io/kubectl/pkg/cmd/create/create_configmap_test.go

```diff
@@ -352,104 +396,130 @@ func TestCreateConfigMap(t *testing.T) {
+			expectErrStr:  `key names or file paths cannot contain '='`,
 		},
 		"create_invalid_configmap_filepath_key_contains_=": {
 			configMapName: "foo",
 			fromFile:      []string{"=key=/file1"},
 			expectErr:     true,
+			expectErrStr:  `key names or file paths cannot contain '='`,
 		},
 		"create_invalid_configmap_literal_key_contains_=": {
 			configMapName: "foo",
 			fromFile:      []string{"=key=value1"},
 			expectErr:     true,
+			expectErrStr:  `key names or file paths cannot contain '='`,
 		},
 		"create_invalid_configmap_duplicate_key1": {
 			configMapName: "foo",
 			fromLiteral:   []string{"key1=value1", "key1=value2"},
 			expectErr:     true,
+			expectErrStr:  `cannot add key "key1", another key by that name already exists in Data for ConfigMap "foo"`,
 		},
 		"create_invalid_configmap_no_file": {
 			configMapName: "foo",
 			fromFile:      []string{"key1=/file1"},
 			expectErr:     true,
+			expectErrStr:  `error reading /file1: no such file or directory`,
 		},
 		"create_invalid_configmap_invalid_literal": {
 			configMapName: "foo",
 			fromLiteral:   []string{"key1value1"},
 			expectErr:     true,
+			expectErrStr:  `invalid literal source key1value1, expected key=value`,
 		},
 		"create_invalid_configmap_invalid_filepath": {
 			configMapName: "foo",
 			fromFile:      []string{"key1==file1"},
 			expectErr:     true,
+			expectErrStr:  `key names or file paths cannot contain '='`,
 		},
 		"create_invalid_configmap_too_many_args": {
 			configMapName: "too_many_args",
 			fromFile:      []string{"key1=/file1"},
-			fromEnvFile:   "foo",
+			fromEnvFile:   []string{"file.env"},
 			expectErr:     true,
+			expectErrStr:  `from-env-file cannot be combined with from-file or from-literal`,
 		},
 		"create_invalid_configmap_too_many_args_1": {
 			configMapName: "too_many_args_1",
 			fromLiteral:   []string{"key1=value1"},
-			fromEnvFile:   "foo",
+			fromEnvFile:   []string{"file.env"},
 			expectErr:     true,
+			expectErrStr:  `from-env-file cannot be combined with from-file or from-literal`,
 		},
 	}
 
 	// run all the tests
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
+			var configMap *corev1.ConfigMap = nil
 			configMapOptions := ConfigMapOptions{
 				Name:           test.configMapName,
 				Type:           test.configMapType,
 				AppendHash:     test.appendHash,
 				FileSources:    test.fromFile,
 				LiteralSources: test.fromLiteral,
-				EnvFileSource:  test.fromEnvFile,
+				EnvFileSources: test.fromEnvFile,
 			}
 
 			if test.setup != nil {
 				if teardown := test.setup(t, &configMapOptions); teardown != nil {
 					defer teardown()
 				}
 			}
+			err := configMapOptions.Validate()
+			if err == nil {
+				configMap, err = configMapOptions.createConfigMap()
+			}
 
-			configMap, err := configMapOptions.createConfigMap()
 			if !test.expectErr && err != nil {
-				t.Errorf("test %s, unexpected error: %v", name, err)
+				t.Errorf("unexpected error: %v", err)
```

Comment:
removed the `name` because if the unit test fails it would already output the name so in my opinion it's repetitive and confusing